keras_nlp/models/bert/bert_tasks.py

```diff
@@ -90,35 +148,48 @@ def __init__(
         super().__init__(
             inputs=inputs,
             outputs=outputs,
+            include_preprocessing=preprocessor is not None,
             **kwargs,
         )
         # All references to `self` below this line
         self._backbone = backbone
+        self._preprocessor = preprocessor
         self.num_classes = num_classes
 
+    def preprocess_samples(self, x, y=None, sample_weight=None):
+        return self.preprocessor(x, y=y, sample_weight=sample_weight)
+
     @property
     def backbone(self):
         """A `keras_nlp.models.Bert` instance providing the encoder submodel."""
         return self._backbone
 
+    @property
+    def preprocessor(self):
+        """A `keras_nlp.models.BertPreprocessor` for preprocessing inputs."""
+        return self._preprocessor
+
     def get_config(self):
         return {
             "backbone": keras.layers.serialize(self.backbone),
+            "preprocessor": keras.layers.serialize(self.preprocessor),
             "num_classes": self.num_classes,
             "name": self.name,
             "trainable": self.trainable,
         }
 
     @classmethod
     def from_config(cls, config):
-        if "backbone" in config:
+        if "backbone" in config and isinstance(config["backbone"], dict):
             config["backbone"] = keras.layers.deserialize(config["backbone"])
+        if "preprocessor" in config and isinstance(
```

Comment:
There is an interesting consequence of rebasing over https://github.com/keras-team/keras-nlp/pull/494.

We can no longer either restore entirely from config or entirely form the main constructor. The backbone needs to be restored via `deserialize`, but the `preprocessor` cannot be restored via `deserialize`, because the vocab need to be downloaded as part of the config (and that is only available via `from_preset`).

I don't think we need to solve this perfectly on this PR, as it more has to with our internals for loading than any actual usage.

If we try to solve the vocabulary loading problem, we could have an interesting setting of config all the way down...

```
"config": {
    "backbone": {
        "class_name": "keras_nlp>Bert",
        "config": {
            "vocabulary_size": 30522,
            "hidden_dim": 128,
            "intermediate_dim": 512,
            "num_layers": 2,
            "num_heads": 2,
            "max_sequence_length": 512,
            "num_segments": 2,
            "dropout": 0.1,
        },
    },
    "preprocessor": {
        "class_name": "keras_nlp>BertPreprocessor",
        "config": {
            "truncator": "waterfall",
            "sequence_length": 128,
        },
    },
    "num_classes": 2,
},
```

This would be interesting in that it would give you to use non-default preprocessing argument during fine-tuning and replicate them here.

Anyway, I propose we just open an issue for this, and track this on a follow up!